### PR TITLE
OctopusCLI packaging misc improvements

### PR DESCRIPTION
**Pull request by thedewi**
# Results

This PR applies a few minor changes to Octopus CLI's linux package deployment scripts, in preparation for them to be adapted to more general scripts.

## Changes

- Rename docker image `bionic-fpm` to `package-linux-docker`: The new name is more future-proof (as ubuntu bionic is no longer current), and more consistent with the name of the other docker image we use later in this process, `publish-linux-docker`.

- Exclude capitalized wrapper instead of deleting: This is to minimize tampering with the input files, and a single option will be easier to refactor than the line of script.

- Give chmod a lighter touch: This script only cares that the main command is executable.

- Introduce variable for Artifactory base URL: Extracts a repeated string into a variable

- Set scripts executable: Sets `+x` in Git on a number of script files that didn't have it.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -359,7 +359,7 @@ Task("CreateLinuxPackages")
             $"{Environment.CurrentDirectory}/artifacts/OctopusTools.{nugetVersion}.linux-x64.extracted:/app",
             $"{Environment.CurrentDirectory}/artifacts:/out"
         }
-    }, "octopusdeploy/bionic-fpm:latest", "bash /build/create-linux-packages.sh");
+    }, "octopusdeploy/package-linux-docker:latest", "bash /build/create-linux-packages.sh");
```
Comment (Waldo000000) [inline build.cake:362]: Right, I see this is in line with this corresponding rename in the project that builds these images: https://build.octopushq.com/admin/settingsDiffView.html?id=buildType:Infrastructure_DockerImages_LinuxPackagerDebRpm&versionBefore=9&versionAfter=10&actionId=475767

File: BuildAssets/create-linux-packages.sh [hunk 2]
```diff
@@ -52,6 +49,7 @@ fpm --version "$VERSION" \
   --depends 'libkrb5-3' \
   --depends 'zlib1g' \
   --depends 'libicu52 | libicu55 | libicu57 | libicu60 | libicu63 | libicu66' \
+  --exclude 'opt/octopus/octopuscli/Octo' \
```
Comment (Waldo000000) [inline BuildAssets/create-linux-packages.sh:52]: I'm a bit curious as to why we need to `Exclude Octo legacy wrapper from distribution` in this script. Is it worth retaining that comment:

> # Exclude Octo legacy wrapper from distribution

...and/or adding extra detail about why the "legacy wrapper" is still built at all?

Comment (thedewi) [reply BuildAssets/create-linux-packages.sh:52]: It's still distributed in the `.tar.gz` for direct download, for compatibility with older versions of that tarball when the command was capitalized.

Comment (thedewi) [reply BuildAssets/create-linux-packages.sh:52]: I will clarify that in a comment.

Comment (Waldo000000) [reply BuildAssets/create-linux-packages.sh:52]: 👍

**Pull request merged**